internal/controlplane/xds_routes.go

```diff
@@ -521,3 +539,11 @@ func hasPublicPolicyMatchingURL(options *config.Options, requestURL url.URL) boo
 	}
 	return false
 }
+
+func isProxyFrontingAuthenticate(options *config.Options, authenticateURL *url.URL, domain string) bool {
```

Comment:
Does this function need both options and `authenticateURL` ?

Could we just grab it from options?

```go
	authenticateURL, err := options.GetAuthenticateURL()
	if err != nil {
		return nil, err
	}
```